lib/SPIRV/SPIRVInternal.h

```diff
@@ -447,6 +447,10 @@ enum Spir2SamplerKind {
   CLK_FILTER_LINEAR = 0x0020,
 };
 
+/// This contains a pair of the pointer element type and an indirection
+/// parameter (to capture cases where an array of OpenCL types is used).
+typedef llvm::PointerIntPair<llvm::Type *, 1, bool> PointerIndirectPair;
```

Comment:
Do we need this duplicate definition?  There is a `#include "LLVMSPIRVLib.h"` at the top already.

Reply:
I don't believe we do. (As historical note, I added it here first, and only added it to `LLVMSPIRVLib.h` after discovering that I needed support for it in `mangleOpenClBuiltin` as well.)